File: algorithms/python3/287_find_the_duplicate_number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in range(len(nums)):
            while nums[i]!=i+1:
                if nums[i]==nums[nums[i]-1]:
                    return nums[i]
                cur=nums[nums[i]-1]
                nums[i],nums[cur]=nums[cur],nums[i]

a=Solution()

nums=[3,1,3,4,2]

print(a.findDuplicate(nums))

File: algorithms/cpp/287_find_the_duplicate_number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        if (nums.empty()) return 0;
        for (int i=0;i<nums.size();i++) {
            while (nums[i]!=i+1) {
                if (nums[i]==nums[nums[i]-1]) return nums[i];
                int temp=nums[nums[i]-1];
                nums[nums[i]-1]=nums[i];
                nums[i]=temp;
            }
            }
        }
};
